modules/openapi-generator/src/main/resources/typescript/api/api.mustache

```diff
@@ -42,8 +45,11 @@ export class {{classname}}RequestFactory extends BaseAPIRequestFactory {
      * @param {{paramName}} {{description}}
      {{/allParams}}
      */
-    public async {{nickname}}({{#allParams}}{{paramName}}{{^required}}?{{/required}}: {{{dataType}}}, {{/allParams}}_options?: Configuration): Promise<RequestContext> {
-        let _config = _options || this.configuration;
+    public async {{nickname}}({{#allParams}}{{paramName}}{{^required}}?{{/required}}: {{{dataType}}}, {{/allParams}}_options?: Configuration{{^useInversify}} | Middleware[]{{/useInversify}}): Promise<RequestContext> {
```

Comment:
could it also be a separate parameter instead of turning _options into a type union? same for all similar changes in the other files

Reply:
Sure, we're okay with users having to pass null/undefined to skip args? I can just put it after _options arg

Reply:
That looks cleaner imo, and easier to consume.
Should the current `configuration` go in there too?

Reply:
that wouldn't be backwards compatible, so i would avoid it